### PR TITLE
Add consoleManagedByDevToolsDuringStrictMode feature flag in React Reconciler

### PR DESCRIPTION
**Pull request by lunaruan**
Right now we don't have support to read the React DevTools settings before initial render on React Native. This PR adds a React feature flag to always disable double logging console logs on React Native until React Native adds support for synchronously reading settings in the future

### CODE REVIEW

File: packages/shared/ReactFeatureFlags.js
Comment (bvaughn) [inline packages/shared/ReactFeatureFlags.js:179]: nit Thinking out loud...I wonder if we could pick a name that also mentions DevTools here?

Like `consoleManagedByDevToolsDuringStrictMode` or something?

I dunno maybe that sucks 😂 I don't feel strongly about this. Feel free to ignore.

**Review by bvaughn — COMMENTED** [review]
I think we should also add a DevTools feature flag to turn off the new setting for standalone build (until React Native supports it)? Otherwise we'll show an option in DevTools that won't make any sense?

**Comment by lunaruan** [comment]
> I think we should also add a DevTools feature flag to turn off the new setting for standalone build (until React Native supports it)? Otherwise we'll show an option in DevTools that won't make any sense?

Yep! I have a feature flag PR ready that I'll put up after the RN builds PR lands!

**Comment by lunaruan** [comment]
> @lunaruan so this pr is for adding it to the reconciler, and the one that already landed (#22215) is for adding it in devtools? is this one good to land now?

yup! Should be pending review!

**Comment by bvaughn** [comment]
> yup! Should be pending review!

That one's been approved, righ?

**Comment by lunaruan** [comment]
Yeah I already landed that one. Waiting for this one to be approved so I can land it too

**Review by bvaughn — APPROVED** [review]

**Comment by gaearon** [comment]
I feel like I don't fully follow what the flag represents and in which case it's supposed to be on/off.

It's currently on for WWW (FB), for Test Renderer (open source and FB). It's currently off for web React (open source), and for RN (open source and FB).

What is the pattern between these? I originally thought it's for RN only (per PR description). But it's off for RN. Then I thought it's meant to be enabled everywhere _except_ RN. But it's on for WWW and Test Renderer. So I'm not sure.

**Comment by gaearon** [comment]
Sent https://github.com/facebook/react/pull/22253#pullrequestreview-747391332 with what trips me up

**Comment by bvaughn** [comment]
> It's currently off for web React (open source)

This was a mistake/oversight.

> It's currently off...for RN (open source and FB).

This was intentional, as DevTools has no way to intercept console logs during initial render / mount in React Native.

**Pull request merged**
